src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts

```diff
@@ -1,46 +1,362 @@
+    }
+
+    while (sliderLength < count) {
+      addButton.click();
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      do {
+        sliderElems = await webdriver.driver.$('.mainContent').$$('.range');
+      } while (sliderElems.length === sliderLength);
+      sliderLength = sliderElems.length;
+    }
+  });
+};
+
+export const whenChangeBandCount = (
+  when: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  when(/^I click to (add|remove) a frequency band$/, async (action: string) => {
+    const isAdd = action === 'add';
+
+    if (isAdd) {
+      const addButton = await webdriver.driver.$(
+        '.mainContent [aria-label="Add Equalizer Slider"]'
+      );
+      addButton.click();
+    } else {
+      const removeButton = await webdriver.driver.$(
+        '.mainContent [aria-label="Remove Equalizer Slider"]'
+      );
+      removeButton.click();
+    }
+
+    // wait 1000 ms for the action.
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  });
+};
+
+// ====================================
+
+const setFrequencyGain = async (
+  webdriver: { driver: Driver | undefined },
+  frequency: number,
+  position: string
+) => {
+  const element = await webdriver.driver.$(
+    `.mainContent input[name="${frequency}-gain-range"]`
+  );
+  const coord = { x: 0, y: 0 };
+  if (position === 'top') {
+    coord.y = -100;
+  } else if (position === 'bottom') {
+    coord.y = 100;
+  }
+  element.dragAndDrop(coord);
+  // wait 1000 ms for the action.
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+};
 
 export const whenSetFrequencyGain = (
   when: DefineStepFunction,
   webdriver: { driver: Driver | undefined }
 ) => {
   when(
-    /^I set gain of slider of frequency (\d+)Hz to (top|middle|bottom)$/,
+    /^I set gain of slider of frequency (\d+)Hz to (top|bottom)$/,
     async (frequency: number, position: string) => {
-      const sliderElems = await webdriver.driver.$('.mainContent').$$('.range');
-      for (let i = 0; i < sliderElems.length; i += 1) {
-        const element = await sliderElems[i].$('input');
-        const name = await element.getAttribute('name');
-        if (name === `${frequency}-gain-range`) {
-          const coord = { x: 0, y: 0 };
-          if (position === 'top') {
-            coord.y = -100;
-          } else if (position === 'bottom') {
-            coord.y = 100;
-          }
-          element.dragAndDrop(coord);
-          // wait 1000 ms for the action.
-          await new Promise((resolve) => setTimeout(resolve, 1000));
-          return;
-        }
+      await setFrequencyGain(webdriver, frequency, position);
+    }
+  );
+};
+
+// ====================================
+
+const setFrequencyQuality = async (
+  webdriver: { driver: Driver | undefined },
+  frequency: number,
+  quality: string
+) => {
+  const element = await webdriver.driver.$(
+    `.mainContent label[for="${frequency}-quality"]`
```

Comment:
is there a reason why we can't just do `.mainContent label[for="${frequency}-quality"] input` directly?